Mathlib/Order/Lattice.lean

```diff
@@ -0,0 +1,1469 @@
+#align sup_lt_iff sup_lt_iff
+
+theorem inf_ind (a b : α) {p : α → Prop} : p a → p b → p (a ⊓ b) :=
+  @sup_ind αᵒᵈ _ _ _ _
+#align inf_ind inf_ind
+
+@[simp]
+theorem inf_le_iff : b ⊓ c ≤ a ↔ b ≤ a ∨ c ≤ a :=
+  @le_sup_iff αᵒᵈ _ _ _ _
+#align inf_le_iff inf_le_iff
+
+@[simp]
+theorem inf_lt_iff : b ⊓ c < a ↔ b < a ∨ c < a :=
+  @lt_sup_iff αᵒᵈ _ _ _ _
+#align inf_lt_iff inf_lt_iff
+
+@[simp]
+theorem lt_inf_iff : a < b ⊓ c ↔ a < b ∧ a < c :=
+  @sup_lt_iff αᵒᵈ _ _ _ _
+#align lt_inf_iff lt_inf_iff
+
+variable (a b c d)
+
+theorem max_max_max_comm : max (max a b) (max c d) = max (max a c) (max b d) :=
+  sup_sup_sup_comm _ _ _ _
+#align max_max_max_comm max_max_max_comm
+
+theorem min_min_min_comm : min (min a b) (min c d) = min (min a c) (min b d) :=
+  inf_inf_inf_comm _ _ _ _
+#align min_min_min_comm min_min_min_comm
+
+end LinearOrder
+
+theorem sup_eq_maxDefault [SemilatticeSup α] [DecidableRel ((· ≤ ·) : α → α → Prop)]
+    [IsTotal α (· ≤ ·)] :
+    (· ⊔ ·) = (maxDefault : α → α → α) := by
+  ext (x y)
+  unfold maxDefault
+  split <;> rename_i h'
+  exacts[sup_of_le_right h', sup_of_le_left $ (total_of (· ≤ ·) x y).resolve_left h']
+#align sup_eq_max_default sup_eq_maxDefault
+
+theorem inf_eq_minDefault [SemilatticeInf α] [DecidableRel ((· ≤ ·) : α → α → Prop)]
+    [IsTotal α (· ≤ ·)] :
+    (· ⊓ ·) = (minDefault : α → α → α) := by
+  ext (x y)
+  unfold minDefault
+  split <;> rename_i h'
+  exacts[inf_of_le_left h', inf_of_le_right $ (total_of (· ≤ ·) x y).resolve_left h']
+#align inf_eq_min_default inf_eq_minDefault
+
+/-- A lattice with total order is a linear order.
+
+See note [reducible non-instances]. -/
+@[reducible]
+def Lattice.toLinearOrder (α : Type u) [Lattice α] [DecidableEq α]
+    [DecidableRel ((· ≤ ·) : α → α → Prop)]
+    [DecidableRel ((· < ·) : α → α → Prop)] [IsTotal α (· ≤ ·)] : LinearOrder α :=
+  { ‹Lattice α› with
+    decidable_le := ‹_›,
+    decidable_eq := ‹_›,
+    decidable_lt := ‹_›,
+    le_total := total_of (· ≤ ·),
+    max := (· ⊔ ·),
+    max_def := by exact congr_fun₂ sup_eq_maxDefault,
+    min := (· ⊓ ·),
+    min_def := by exact congr_fun₂ inf_eq_minDefault }
+#align lattice.to_linear_order Lattice.toLinearOrder
+
+-- see Note [lower instance priority]
+instance (priority := 100) LinearOrder.toDistribLattice {α : Type u} [LinearOrder α] :
+  DistribLattice α :=
+  { LinearOrder.toLattice with
+    le_sup_inf := fun _ b c =>
+      match le_total b c with
+      | Or.inl h => inf_le_of_left_le $ sup_le_sup_left (le_inf (le_refl b) h) _
+      | Or.inr h => inf_le_of_right_le $ sup_le_sup_left (le_inf h (le_refl c)) _ }
+#align linear_order.to_distrib_lattice LinearOrder.toDistribLattice
+
+instance Nat.distribLattice : DistribLattice ℕ := by infer_instance
+#align nat.distrib_lattice Nat.distribLattice
+
+/-! ### Dual order -/
+
+
+open OrderDual
+
+@[simp]
+theorem of_dual_inf [HasSup α] (a b : αᵒᵈ) : ofDual (a ⊓ b) = ofDual a ⊔ ofDual b :=
```

Comment:
Do we use `of_dual` or `ofDual` in lemma names?

Reply:
this would be `ofDual_inf`